cardano-tx-submit/testing-tx-submit.md

```diff
@@ -0,0 +1,73 @@
+# Testing cardano-tx-submit-webapi
+
+Setting this up for testing and for actual use on a real network is significantly more difficult
+than it should be. This is definitely not ready for end users yet.
+
+The biggest thing missing at the moment is documentation of generation of transactions with the
+command line tools. That needs to be *thoroughly* documented and this document is not the place
+for that.
+
+The following instructions are for generating a testnet and testing tx-submission on that testnet.
+
+* This process generates a lot of state in the `cardano-node` checkout that can be cleaned up using:
+    ```
+    cardano-node $ rm -rf configuration/GenesisFiles/ db-* logs/ transaction.bin
+    ```
+
+* Generated the Genesis file for the testnet:
+    ```
+    cardano-node $ ./scripts/genesis.sh
```

Comment:
If I am not mistaken I discovered 'fun' fact today. If your root directory will have spaces in it the execution of this script will break, I need to recheck it later but had this error:

>  ++ cabal v2-run -v0 -- cardano-cli --log-config '/home/artur/Projects/Explorer Readme/cardano-node/configuration/configuration-silent.yaml' --real-pbft print-genesis-hash --genesis-json /tmp/tmp.HAVPtFP5k8.d/genesis.json
> ++ tail -1
> warning: redirecting to https://github.com/well-typed/canonical-json/
> remote: Enumerating objects: 1, done.
> remote: Counting objects: 100% (1/1), done.
> remote: Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
> Unpacking objects: 100% (1/1), done.
> From https://github.com/input-output-hk/cardano-ledger-specs
>    df2bea65..4f318ea4  master     -> origin/master
> + GENHASH=d86c1f4e32a459a38c4fb07d097a7a2771f1ea502e801bd297ff08c3d7594cb4
> + TARGETDIR='/home/artur/Projects/Explorer Readme/cardano-node/configuration/genesis'
> + mkdir -vp '/home/artur/Projects/Explorer Readme/cardano-node/configuration/genesis'
> mkdir: created directory '/home/artur/Projects/Explorer Readme/cardano-node/configuration/genesis'
> + cp -iav /tmp/tmp.HAVPtFP5k8.d/genesis.json /home/artur/Projects/Explorer Readme/cardano-node/configuration/genesis/
> cp: target 'Readme/cardano-node/configuration/genesis/' is not a directory

When I changed it to `Explorer Readme` - without spaces it ended OK

Reply:
I rechecked it and it happened again, I opened ticket for that: https://github.com/input-output-hk/cardano-node/issues/412

Reply:
Directory names with spaces are always a mistake :).